### PR TITLE
Fix/find 541 return 404 for pages over 500

### PR DESCRIPTION
**Pull request by TNA-Allan**
# Pull Request

Ticket URL: [FIND-541](https://national-archives.atlassian.net/browse/FIND-541)

## About these changes

- Addresses pagination behaviour around the Elasticsearch (ES) 10,000-record query limit. Requests for pages above PAGE_LIMIT now consistently return 404 without querying the API, instead of potentially resulting in a 500 from a downstream Bad Request. This also aligns behaviour for requests just beyond the limit, which previously returned empty results.
- Replaces hard-coded page limit warning values and messages with constants, keeping pagination-related configuration in one place.
- Adds test coverage for pagination boundary conditions and page limit handling.
- Documents the ES 10,000-record limitation.
- Add comments for clarity.

## How to check these changes

http://localhost:65533/catalogue/search/?page=501 - shows Page Not Found - 404 (previously No Results Found)
http://localhost:65533/catalogue/search/?page=999999999 - shows Page Not Found - 404 (previously 500 Service Error)

Regression:
http://localhost:65533/catalogue/search/?page=abc - show Page Not Found - 404
http://localhost:65533/catalogue/search/?page=0 - show Page Not Found - 404
http://localhost:65533/catalogue/search/?page=1
http://localhost:65533/catalogue/search/?page=495 Page limit warning threshold - show warning message
http://localhost:65533/catalogue/search/?page=500 last page and above Page limit warning threshold - show warning message
http://localhost:65533/catalogue/search/?q=ufo&page=35 last page within page limit and warning threshold does not show warning message
http://localhost:65533/catalogue/search/?q=qwert - shows No results found

## Before assigning to reviewer, please make sure you have

- Checked things thoroughly before handing over to reviewer
- Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- Ensured main branch is deployable and all non-live features are behind flags
- Ensured that PR includes only commits relevant to the ticket
- Waited for all CI jobs to pass before requesting a review
- Added/updated tests and documentation where relevant
- Ensured the merge is unblocked (e.g., all commits have verified signatures)

[FIND-541]: https://national-archives.atlassian.net/browse/FIND-541?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/backend/search-limitations.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- **RESULTS_PER_PAGE**: 20 results per page
+- **PAGE_LIMIT**: 500 pages maximum
+- **Math**: 500 pages × 20 results/page = 10,000 results
+
+See [app/search/constants.py](../../app/search/constants.py) for these constants.
+
+### What Happens When Limits Are Hit
+
+- **Invalid page number** (< 1, not an integer, or > PAGE_LIMIT): View raises `PageNotFound`, returns HTTP 404 immediately without querying the API.
+- **Page beyond calculated total**: If results total fewer than 10,000, requesting a page beyond the calculated total also returns HTTP 404.
```
[the remark this answers is not in the document]
Comment (TNA-Allan) [reply docs/backend/search-limitations.md:20]: 0 results query is bypassed by Invalid page number.

**Review by Firdaussi — APPROVED** [review]
Looks good
